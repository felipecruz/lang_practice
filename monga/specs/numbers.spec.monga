int a;

void fun () {
    a = 1;
    a = 2;
    a = 3;
    a = 100;
    a = 100000;
    a = 100000000;
    a = 0xFFFFFFF;

    a = 0x00;
    a = 0x01;
    a = 0x02;
    a = 0x03;
    a = 0x04;
    a = 0x05;
    a = 0x06;
    a = 0x06;
    a = 0x07;
    a = 0x08;
    a = 0x0A;
    a = 0x0B;
    a = 0x0C;
    a = 0x0D;
    a = 0x0E;
    a = 0x0F;

    a = 0x10;
    a = 0x11;
    a = 0x12;
    a = 0x13;
    a = 0x14;
    a = 0x15;
    a = 0x16;
    a = 0x16;
    a = 0x17;
    a = 0x18;
    a = 0x1A;
    a = 0x1B;
    a = 0x1C;
    a = 0x1D;
    a = 0x1E;
    a = 0x1F;

    a = 0x20;
    a = 0x21;
    a = 0x22;
    a = 0x23;
    a = 0x24;
    a = 0x25;
    a = 0x26;
    a = 0x26;
    a = 0x27;
    a = 0x28;
    a = 0x2A;
    a = 0x2B;
    a = 0x2C;
    a = 0x2D;
    a = 0x2E;
    a = 0x2F;

    a = 0x30;
    a = 0x31;
    a = 0x32;
    a = 0x33;
    a = 0x34;
    a = 0x35;
    a = 0x36;
    a = 0x36;
    a = 0x37;
    a = 0x38;
    a = 0x3A;
    a = 0x3B;
    a = 0x3C;
    a = 0x3D;
    a = 0x3E;
    a = 0x3F;
}
