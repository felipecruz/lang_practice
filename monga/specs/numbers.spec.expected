
TK_NUMBER: 1

TK_NUMBER: 2

TK_NUMBER: 3

TK_NUMBER: 100

TK_NUMBER: 100000

TK_NUMBER: 100000000

TK_HEXA: 0xFFFFFFF



TK_HEXA: 0x0        

TK_HEXA: 0x1        

TK_HEXA: 0x2        

TK_HEXA: 0x3        

TK_HEXA: 0x4        

TK_HEXA: 0x5        

TK_HEXA: 0x6        

TK_HEXA: 0x6        

TK_HEXA: 0x7        

TK_HEXA: 0x8        

TK_HEXA: 0xA        

TK_HEXA: 0xB        

TK_HEXA: 0xC        

TK_HEXA: 0xD        

TK_HEXA: 0xE        

TK_HEXA: 0xF        


TK_HEXA: 0x10        

TK_HEXA: 0x11        

TK_HEXA: 0x12        

TK_HEXA: 0x13        

TK_HEXA: 0x14        

TK_HEXA: 0x15        

TK_HEXA: 0x16        

TK_HEXA: 0x16        

TK_HEXA: 0x17        

TK_HEXA: 0x18        

TK_HEXA: 0x1A        

TK_HEXA: 0x1B        

TK_HEXA: 0x1C        

TK_HEXA: 0x1D        

TK_HEXA: 0x1E        

TK_HEXA: 0x1F        


TK_HEXA: 0x20        

TK_HEXA: 0x21        

TK_HEXA: 0x22        

TK_HEXA: 0x23        

TK_HEXA: 0x24        

TK_HEXA: 0x25        

TK_HEXA: 0x26        

TK_HEXA: 0x26        

TK_HEXA: 0x27        

TK_HEXA: 0x28        

TK_HEXA: 0x2A        

TK_HEXA: 0x2B        

TK_HEXA: 0x2C        

TK_HEXA: 0x2D        

TK_HEXA: 0x2E        

TK_HEXA: 0x2F        


TK_HEXA: 0x30        

TK_HEXA: 0x31        

TK_HEXA: 0x32        

TK_HEXA: 0x33        

TK_HEXA: 0x34        

TK_HEXA: 0x35        

TK_HEXA: 0x36        

TK_HEXA: 0x36        

TK_HEXA: 0x37        

TK_HEXA: 0x38        

TK_HEXA: 0x3A        

TK_HEXA: 0x3B        

TK_HEXA: 0x3C        

TK_HEXA: 0x3D        

TK_HEXA: 0x3E        

TK_HEXA: 0x3F        
Valid file
